Proposal: univeresaldot.me

JosephKnecht-lab opened the pull request:
### Project Abstract

[Universaldot.me](http://universaldot.me/) is a freelancing decentralized application (dApp) that allows individuals to create/complete tasks and organize themselves by creating geographically distributed organizations. As such, it intends to simplify the way people and organizations create contracts between one another. The layer of trust and transparency is provided by distributed blockchain infrastructure (Substrate).

> This application is a follow up grant to: https://github.com/w3f/Grants-Program/pull/714

### For which grant [level](https://github.com/w3f/Grants-Program#level_slider-levels) are you applying? 
- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [ ] **Level 2**:  Up to $50,000, 3 approvals
- [x] **Level 3**:  Unlimited, 5 approvals (for > $100k Web3 Foundation Council approval)

### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied, renamed ( `project_name.md`) and updated.
- [x] A BTC or Ethereum (DAI/USDT) address for the payment of the milestones is provided inside the application.
- [x] I have read and acknowledged the [terms and conditions](https://github.com/w3f/Grants-Program/blob/master/docs/T&Cs.md).
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process).

### How Did You Hear About our grants program?

- [ ] Social Media
- [ ] Hackathon
- [ ] Personal Recommendation
- [ ] Substrate Builders Program
- [ ] Investor/VC
- [ ] Online Search 
- [x] Other: WEB3 Foundation website

semuelle commented:
Hi @JosephKnecht-lab, thank you for your follow-up application. I'm going to share your application with the rest of the team already.

semuelle reviewed (changes requested):
A couple of questions from my side:

- We have already funded a number of Substrate-IPFS integrations. Can you elaborate why/how the existing solutions don't work for you?
- I don't understand how Tensorflow relates to the rest of the system. In the architecture diagram, there are `Pallets` blocks and a `Tensorflow` block, but no explanation of how they interact with another. 
- Several of your deliverables are labeled as "redesign" or "improve", but without an explanation why the mentioned elements need to be redesigned and how. Given the size of the grant, I would like to see a lot more detail on that.

You can put the answer to these questions directly into the application document.

JosephKnecht-lab commented:
Hi @semuelle 

We have provided answers to your questions in the section `Aditional Questions` at the end of the document. 

Let us know if you need more details.

semuelle commented:
Thanks for the update, @JosephKnecht-lab. Can you explain how the IPFS integration needs to be customised to the use case? AFAIK, there are existing general-purpose IPFS workers.

Other than that, I think the Tensorflow integration is too niche and the deliverables do not justify 30 person-months. I'll let my colleagues chime in, though.

Noc2 reviewed (changes requested):
Thanks for the application. I agree with Sebastian regarding the IPFS integration. You should take a look at the existing implementations and see if you can potentially leverage this, see for example https://github.com/DanHenton/pocket-substrate/tree/ipfs-ocw We would only support it via grants, if you can specifically detail what you need to do. Also could you provide more information regarding the changes of the front-end? For example, do you have mock-ups for the changes etc.? And regarding Tensorflow how exactly will this be integrated/interact with the substrate chain? Or will this only work with the front-end? 
That said, I think improving the pallets Profile,Task, Dao, given the level of detail you provided would be something we would already be willing to support.

hakan-w3f commented:
I find the ML part of this project super valuable. Is the majority of the ML related work going to be integrating the ML stack with the rest of the architecture? I'm asking this because there are really good off-the-shelf recommendation models for TF.

JosephKnecht-lab commented:
Hi @semuelle , @Noc2 , @hakan-w3f 

We have updated the original application with additional questions and answers. You can find them in the `Aditional Questions`  Section.

hakan-w3f reviewed (dismissed):
This will be the first of its kind among our W3F grantees, and with an interesting use-case. I'll go ahead with one approval.

Noc2 reviewed (changes requested):
Thanks for the update and the additional information. I have a few more questions/comments: Could you remove “Launch Mainnet with custom genesis configuration” and potentially the testnet from the deliverables? We usually don’t support the deployment via our grants program, since this is something that is mostly beneficial for you and your project and not for the entire ecosystem (like the open source code). Also would you be willing to reduce the price a little bit more? In my opinion, if you don’t plan to become a common good chain, more and more of the development should be covered by VCs, since our grants program also not necessarily looks into the business model etc.  Btw. we would be more than happy to help you with finding potential VCs in case you are interested in this.

semuelle reviewed (commented):
I like the project in its current form and would like to see it expand. However, I don't support the application in its current form. I don't think the Tensorflow integration is useful for the ecosystem as it's a centralised service serving one frontend and one very specific use case and does not address any critical functionality. Serving suggestions based on keywords would achieve the same. The IPFS integration and pallet redesigns do not justify 27 person-months.

JosephKnecht-lab commented:
I have addressed your comments in the application document again. 

Let us know how to proceed further. 

FYI: we have moved the previous deployment of the dApp to a new endpoint: https://app.universaldot.me

hakan-w3f reviewed (dismissed):
@JosephKnecht-lab, thanks for the update. Re-approving. Cheers.

Noc2 reviewed (dismissed):
Thanks for the update. I’m happy to go ahead with it, but after this grant the project should try to get funded via VCs or treasury (in the case of common good) and potentially also join the SBP.

takahser reviewed (changes requested):
In addition to my comments, I'm curious if you have something like a class diagram that provides a brief overview of the concepts you're using (e.g. task, organisation, dao, etc.) and how they relate to each other (e.g. organisation can have multiple tasks, etc.).

JosephKnecht-lab commented on applications/universaldot.me.md:
```diff
@@ -0,0 +1,292 @@
+- The dApp eventually aims to become a parachain on polkdot. 
+
+*Who is your target audience (parachain/dapp/wallet/UI developers, designers, your own user base, some dapp's userbase, yourself)?*
+- Users who want to create decentralized organizations and projects. 
```
Yes, conceptually it is similar to decentralized Jira. Organizations can manage a Kanban Board of `Tasks`, and these tasks will be suggested to users based on their `Profile` information. 

In our current design, `Task` is very generically implemented. This means that a task can be something like:

- take me from place A to place B (example in car-sharing)
- create web design for our company. 
- create a decentralized Jira. 

The creator of the `Tasks` basically chooses the level of abstraction for his task. 

We see the `DAO` as a collection of `Tasks`, which fulfill the organizational vision. As such, for now, we don't intend to implement multiple projects inside a DAO. It might be needed in the future if the market demands this new abstraction.

JosephKnecht-lab commented:
Thanks for the suggestions @takahser.

I have updated the application based on your suggestions and addressed your general comment in the application document itself.
